src/shared/User/api.js

```diff
@@ -1,8 +1,26 @@
-import { getClient, checkResponse } from 'shared/Swagger/api';
+import Swagger from 'swagger-client';
+
+import { getClient, checkResponse, requestInterceptor } from 'shared/Swagger/api';
 
 export async function GetLoggedInUser() {
   const client = await getClient();
   const response = await client.apis.users.showLoggedInUser({});
   checkResponse(response, 'failed to get user due to server error');
   return response.body;
 }
+
+export function LogoutUser() {
+  const logoutEndpoint = '/auth/logout';
+  const req = {
+    url: logoutEndpoint,
+    method: 'POST',
+    requestInterceptor,
```

Comment:
Found the issue! The `_gorilla_csrf` cookie wasn't sent in the request. `Swagger.http()` doesn't automatically send the cookies in the requests. You have to pass in `credentials: 'same-origin'` to let swagger know to send the cookies with the request.

```suggestion
    credentials: 'same-origin',
    requestInterceptor,
```

Reply:
More info here: https://github.com/swagger-api/swagger-ui/issues/2744

Reply:
This makes a ton of sense and I'm super impressed you found this. Two things this made me remember:

1) `Swagger.http` is using `fetch` under the hood unless you specify something different.  Here's the fetch documentation on what you shared with me: https://github.com/github/fetch#sending-cookies

2) If Gorilla CSRF doesn't come back with the `_gorilla_csrf` cookie then I noticed it will re-create the cookie.  That will then fail.  And its exactly what I was seeing, the failure was that the cookies never matched.  Here's the code for that: https://github.com/gorilla/csrf/blob/master/csrf.go#L193-L213

I'll update this and try it out on Edge.

Reply:
This worked in Edge!